File: Data Structures & Algorithms/majority-element/submission-1.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        hash_map = defaultdict(int)
        max_val = 0
        for item in nums:
            hash_map[item] += 1
            max_val = max(max_val, hash_map[item])

        for key in hash_map.keys():
            if hash_map[key] == max_val:
                return key
        

  

File: Data Structures & Algorithms/majority-element-ii/submission-1.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        int threshold = nums.size() / 3;
        
        unordered_map<int,int> my_map;
        vector<vector<int>> freq(nums.size()+1);
        vector<int> res;



        for (int i=0; i<nums.size(); i++) {
            my_map[nums[i]]++;
        }

        for (auto & [key, value] : my_map) {
            freq[value].push_back(key);
        }

        for (int i=freq.size()-1; i > threshold; i--) {
            for (int item : freq[i]) {
                res.push_back(item);
            }
        }

        return res;

    }
};